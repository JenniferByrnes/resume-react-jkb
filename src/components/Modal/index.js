import React from 'react'

function Modal({ onClose, currentPhoto }) {
  const {name, category, description, index} = currentPhoto;

  return (
    <div className="modalBackdrop">
      <div className="modalContainer">
        <h3 className="modalTitle">{name}</h3>
        <img src={require(`../../assets/projects/${index}.jpg`)} alt="current category" />
        <p>{description}</p>
          <button onClick={onClose} type="button">Close this modal</button>
        </div>
      </div>
    );
  }

export default Modal;