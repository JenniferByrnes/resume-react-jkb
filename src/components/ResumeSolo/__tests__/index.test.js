import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const currentPhoto = {
  name: 'Park bench',
  category: 'landscape',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
  index: 1
};

const mockToggleModal = jest.fn();

afterEach(cleanup);

describe('Modal component renders', () => {
  // baseline test
  it('renders', () => {
    render(<Modal
    currentPhoto = {currentPhoto}
    toggleModal={mockToggleModal}
    />);
  });

  it('renders', () => {
    //snapshot test
    const { asFragment } = render(<Modal 
      currentPhoto = {currentPhoto}
      toggleModal={mockToggleModal}/>)
    expect(asFragment()).toMatchSnapshot()
  })
})

it('renders', () => {
  const { getByRole } = render(<Modal 
    currentPhoto = {currentPhoto}
    toggleModal={mockToggleModal}/>)
  expect(getByRole('button')).toHaveTextContent('Close this modal')
})

describe('Click Event', () => {
  it('calls onClose handler', () => {
    // Arrange: Render Modal
    const { getByText } = render(<Modal
      onClose={mockToggleModal}
      currentPhoto={currentPhoto}
    />);
    // Act: Simulate click event
    fireEvent.click(getByText('Close this modal'));
    // Assert: Expected matcher
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
})  


