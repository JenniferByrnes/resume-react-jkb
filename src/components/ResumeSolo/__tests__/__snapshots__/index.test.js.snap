// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders renders 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
